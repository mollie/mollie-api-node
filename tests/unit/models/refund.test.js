import Refund from 'models/refund';

describe('refund model', () => {
  it('should instantiate with default values', () => {
    const refund = new Refund();

    expect(refund.isQueued()).toBe(false);
    expect(refund.isPending()).toBe(false);
    expect(refund.isProcessing()).toBe(false);
    expect(refund.isRefunded()).toBe(false);
    expect(refund.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const refundProps = JSON.parse(`{
        "resource": "refund",
        "id": "re_4qqhO89gsT",
        "amount": {
            "currency": "EUR",
            "value": "5.95"
        },
        "status": "pending",
        "createdAt": "2018-03-14T17:09:02.0Z",
        "description": "Order",
        "paymentId": "tr_WDqYK6vllg",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/refunds/re_4qqhO89gsT",
                "type": "application/hal+json"
            },
            "payment": {
                "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
                "type": "application/hal+json"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/refunds-api/get-refund",
                "type": "text/html"
            }
        }
    }`);
    const refund = new Refund(refundProps);

    expect(refund.isQueued()).toBe(false);
    expect(refund.isPending()).toBe(true);
    expect(refund.isProcessing()).toBe(false);
    expect(refund.isRefunded()).toBe(false);
    expect(refund.toPlainObject()).toMatchSnapshot();
  });
});
