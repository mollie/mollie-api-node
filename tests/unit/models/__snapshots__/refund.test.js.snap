// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refund model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "payment": null,
    "settlement": null,
  },
  "amount": Object {
    "currency": null,
    "value": null,
  },
  "createdAt": null,
  "description": null,
  "id": null,
  "paymentId": null,
  "resource": "refund",
  "settlementAmount": null,
  "status": null,
}
`;

exports[`refund model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/refunds-api/get-refund",
      "type": "text/html",
    },
    "payment": Object {
      "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
      "type": "application/hal+json",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/refunds/re_4qqhO89gsT",
      "type": "application/hal+json",
    },
  },
  "amount": Object {
    "currency": "EUR",
    "value": "5.95",
  },
  "createdAt": "2018-03-14T17:09:02.0Z",
  "description": "Order",
  "id": "re_4qqhO89gsT",
  "paymentId": "tr_WDqYK6vllg",
  "resource": "refund",
  "settlementAmount": null,
  "status": "pending",
}
`;
